### PR TITLE
Update the builder image

### PR DESCRIPTION
**Pull request by tenzen-y**
Signed-off-by: tenzen-y <yuki.iwai.tz@gmail.com>

#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:
As described in [this](https://github.com/kubernetes-sigs/kueue/pull/375#issuecomment-1239521085), I updated the builder image.

#### Which issue(s) this PR fixes:

Fixes #

### CODE REVIEW

File: cloudbuild.yaml [hunk 1]
```diff
@@ -4,7 +4,7 @@ timeout: 3000s
 # For each build step, Prow executes a job.
 steps:
 # see https://github.com/kubernetes/test-infra/tree/master/config/jobs/image-pushing
-  - name: gcr.io/k8s-staging-test-infra/gcb-docker-gcloud:v20211013-1be7868d8b
+  - name: gcr.io/k8s-testimages/gcb-docker-gcloud:v20220830-45cbff55bc
     entrypoint: make
     args:
       - image-build
```
Comment (alculquicondor) [inline cloudbuild.yaml:10]: we can remove this line now, right?

`image-push` now also builds

Comment (tenzen-y) [reply cloudbuild.yaml:10]: Yes, that's right. Thanks.

Comment (tenzen-y) [reply cloudbuild.yaml]: Done.

**Comment by tenzen-y** [comment]
We faced the same error...

> unknown flag: --use
See 'docker --help'.

https://storage.googleapis.com/kubernetes-jenkins/logs/post-kueue-push-images/1567573072498659328/build-log.txt

**Comment by alculquicondor** [comment]
This works from a local machine:

```
docker run --rm -ti gcr.io/k8s-testimages/gcb-docker-gcloud:v20220830-45cbff55bc
bash-5.1# docker buildx create --use
youthful_cannon
```

So maybe there is something wrong in the Makefile

**Comment by tenzen-y** [comment]
> This works from a local machine:
> 
> ```
> $ docker run --rm -ti gcr.io/k8s-testimages/gcb-docker-gcloud:v20220830-45cbff55bc
> bash-5.1# docker buildx create --use
> youthful_cannon
> ```
> 
> So maybe there is something wrong in the Makefile

Thanks for sharing.
I will check Makefile.

**Comment by tenzen-y** [comment]
@alculquicondor 
Maybe we need to use `/buildx-entrypoint` instead of make command like `kubernetes/test-infra` and `kubernetes/cloud-provider-aws`.

What do you think about this?

- https://github.com/kubernetes/test-infra/blob/372b1c89983864c5525df6774e56a0fd7287c29e/images/git/cloudbuild.yaml
- https://github.com/kubernetes/cloud-provider-aws/blob/f6b92bb1cd106956d67b5989745eabf663432540/cloudbuild.yaml

**Comment by alculquicondor** [comment]
I was about to comment the same thing.

I still would like to invoque the build process through make. So maybe you can replace `IMAGE_BUILD_CMD` with `/buildx-entrypoint` in `cloudbuild.yaml` file. But then you need to separate the `docker buildx create` into a separate make rule for developers to use in their local machines.

**Pull request merged**